jsapp/js/router/router.tsx

```diff
@@ -0,0 +1,361 @@
+                />
+              }
+            />
+            <Route
+              path={ROUTES.FORM_SHARING}
+              element={
+                <PermProtectedRoute
+                  protectedComponent={FormSubScreens}
+                  requiredPermissions={[PERMISSIONS_CODENAMES.manage_asset]}
+                />
+              }
+            />
+            <Route
+              path={ROUTES.FORM_RECORDS}
+              element={
+                <PermProtectedRoute
+                  protectedComponent={FormSubScreens}
+                  requiredPermissions={[PERMISSIONS_CODENAMES.manage_asset]}
+                />
+              }
+            />
+            <Route
+              path={ROUTES.FORM_REST}
+              element={
+                <PermProtectedRoute
+                  protectedComponent={FormSubScreens}
+                  requiredPermissions={[
+                    PERMISSIONS_CODENAMES.change_asset,
+                    PERMISSIONS_CODENAMES.view_submissions,
+                  ]}
+                  requireAll
+                />
+              }
+            />
+            <Route
+              path={ROUTES.FORM_REST_HOOK}
+              element={
+                <PermProtectedRoute
+                  protectedComponent={FormSubScreens}
+                  requiredPermissions={[PERMISSIONS_CODENAMES.manage_asset]}
+                />
+              }
+            />
+            <Route
+              path={ROUTES.FORM_KOBOCAT}
+              element={
+                <PermProtectedRoute
+                  protectedComponent={FormSubScreens}
+                  requiredPermissions={[PERMISSIONS_CODENAMES.manage_asset]}
+                />
+              }
+            />
+          </Route>
+
+          <Route
+            path={ROUTES.FORM_JSON}
+            element={
+              <PermProtectedRoute
+                protectedComponent={FormJson}
+                requiredPermissions={[PERMISSIONS_CODENAMES.view_asset]}
+              />
+            }
+          />
+          <Route
+            path={ROUTES.FORM_XFORM}
+            element={
+              <PermProtectedRoute
+                protectedComponent={FormXform}
+                requiredPermissions={[PERMISSIONS_CODENAMES.view_asset]}
+              />
+            }
+          />
+          <Route
+            path={ROUTES.FORM_EDIT}
+            element={
+              <PermProtectedRoute
+                protectedComponent={FormPage}
+                requiredPermissions={[PERMISSIONS_CODENAMES.view_asset]}
+              />
+            }
+          />
+          {/**
+           * TODO change this HACKFIX to a better solution
+           *
+           * Used to force refresh form sub routes. It's some kine of a weird
+           * way of introducing a loading screen during sub route refresh.
+           * See: https://github.com/kobotoolbox/kpi/issues/3925
+           **/}
```

Comment:
As mentioned on Zulip, we can drop this 🎉 

> I've looked at that `/forms/<uid>/reset` route. It is being used only in `triggerRefresh()` function from `formViewTabs.es6`. The idea was that when user clicks a menu item, e.g. "Table" from "DATA", the route would reload itself. This is interesting, but since I broke it (of course it was me) over 2 years ago (see https://github.com/kobotoolbox/kpi/commit/f14005c9bdeabd9246e0267e637e60772c57a4fb#diff-cd8026c65e0451326d9e9eb4de92a70018810bf708c2e70affcdab16773a93b1L251) it stopped working at all

Reply:
FORM_KOBOCAT should be removed while we're here, it's superceded by the feature at `/forms/:uid/settings/media`